tests/test_tools.py

```diff
@@ -16,109 +16,201 @@ def _setup_dirs(monkeypatch, user_filesystem):
+    (
+        {"cli_username": None, "cli_email": None, "skip_config_creation": True},
+        {"expected_username": "", "expected_email": ""},
+    ),
+    (
+        {"cli_username": "cli_username", "cli_email": None, "skip_config_creation": True},
+        {"expected_username": "cli_username", "expected_email": ""},
+    ),
+    (
+        {"cli_username": None, "cli_email": "cli@email.com", "skip_config_creation": True},
+        {"expected_username": "", "expected_email": "cli@email.com"},
+    ),
+    (
+        {"cli_username": "cli_username", "cli_email": "cli@email.com", "skip_config_creation": True},
+        {"expected_username": "cli_username", "expected_email": "cli@email.com"},
+    ),
 ]
 
 
-@pytest.mark.parametrize("inputs, expected", params_user_info_with_home_conf_file)
-def test_get_user_info_with_home_conf_file(monkeypatch, inputs, expected, user_filesystem):
+@pytest.mark.parametrize("cli_inputs, expected", params_user_info_with_home_conf_file)
+def test_get_user_info_with_home_conf_file(monkeypatch, cli_inputs, expected, user_filesystem):
     _setup_dirs(monkeypatch, user_filesystem)
-    _run_tests(inputs, expected)
+    _run_tests(cli_inputs, expected)
 
 
-@pytest.mark.parametrize("inputs, expected", params_user_info_with_local_conf_file)
-def test_get_user_info_with_local_conf_file(monkeypatch, inputs, expected, user_filesystem):
+@pytest.mark.parametrize("cli_inputs, expected", params_user_info_with_local_conf_file)
+def test_get_user_info_with_local_conf_file(monkeypatch, cli_inputs, expected, user_filesystem):
     _setup_dirs(monkeypatch, user_filesystem)
     local_config_data = {"username": "cwd_username", "email": "cwd@email.com"}
     with open(Path(user_filesystem) / "diffpyconfig.json", "w") as f:
         json.dump(local_config_data, f)
-    _run_tests(inputs, expected)
+    _run_tests(cli_inputs, expected)
+    # Run tests again without global config, results should be the same
     os.remove(Path().home() / "diffpyconfig.json")
-    _run_tests(inputs, expected)
+    _run_tests(cli_inputs, expected)
 
 
-@pytest.mark.parametrize("inputsa, inputsb, expected", params_user_info_with_no_home_conf_file)
-def test_get_user_info_with_no_home_conf_file(monkeypatch, inputsa, inputsb, expected, user_filesystem):
+@pytest.mark.parametrize("cli_inputs, inputs, expected", params_user_info_with_no_conf_file)
+def test_get_user_info_with_no_conf_file(monkeypatch, cli_inputs, inputs, expected, user_filesystem):
     _setup_dirs(monkeypatch, user_filesystem)
     os.remove(Path().home() / "diffpyconfig.json")
-    inp_iter = iter(inputsb)
+    inp_iter = iter([inputs["input_username"], inputs["input_email"]])
     monkeypatch.setattr("builtins.input", lambda _: next(inp_iter))
-    _run_tests(inputsa, expected)
+    _run_tests(cli_inputs, expected)
     confile = Path().home() / "diffpyconfig.json"
-    assert confile.is_file()
+    assert confile.is_file() == expected["config_file_exists"]
```

Comment:
global config file should only be created when both username and email are given